analytics/pubstack/pubstack_module_test.go

```diff
@@ -105,20 +107,24 @@ func TestPubstackModuleSuccess(t *testing.T) {
 
 		// create server with root endpoint that returns the current config
 		// add an intake endpoint that PBS hits when events are sent
-		rootCount := 0
+		var isFirstQuery bool = true
+		var isFirstQueryMutex sync.Mutex
 		mux := http.NewServeMux()
 		mux.HandleFunc("/", func(res http.ResponseWriter, req *http.Request) {
-			rootCount++
+			isFirstQueryMutex.Lock()
+			defer isFirstQueryMutex.Unlock()
```

Comment:
The `rootCount` variable was not memory protected before and I suspect that's the cause of the failures. The superfluous writes are just warnings and were a red herring.